src/status_im/contexts/wallet/create_account/new_keypair/keypair_name/view.cljs

```diff
@@ -0,0 +1,47 @@
+(ns status-im.contexts.wallet.create-account.new-keypair.keypair-name.view
+  (:require
+    [clojure.string :as string]
+    [native-module.core :as native-module]
+    [quo.core :as quo]
+    [react-native.core :as rn]
+    [reagent.core :as reagent]
+    [status-im.contexts.wallet.create-account.new-keypair.keypair-name.style :as style]
+    [utils.i18n :as i18n]
+    [utils.re-frame :as rf]))
+
+(defn view
+  []
+  (let [keypair-name (reagent/atom "")]
+    (fn []
+      (let [{:keys [secret-phrase]} (rf/sub [:wallet/ui])]
+        [rn/view {:style {:flex 1}}
+         [quo/page-nav
+          {:icon-name           :i/arrow-left
+           :on-press            #(rf/dispatch [:navigate-back])
+           :accessibility-label :top-bar}]
+         [quo/text-combinations
+          {:container-style style/header-container
+           :title           (i18n/label :t/keypair-name)
+           :description     (i18n/label :t/keypair-name-description)}]
+         [quo/input
+          {:container-style {:margin-horizontal 20}
+           :placeholder     (i18n/label :t/keypair-name-input-placeholder)
+           :label           (i18n/label :t/keypair-name)
+           :char-limit      15
+           :on-change-text  #(reset! keypair-name %)}]
+         [quo/bottom-actions
+          {:actions          :one-action
+           :button-one-label (i18n/label :t/continue)
+           :button-one-props {:disabled?           (or (zero? (count @keypair-name))
+                                                       (> (count @keypair-name) 15))
+                              :customization-color :blue
+                              :on-press            #(native-module/create-account-from-mnemonic
```

Comment:
I think we should instead do this in a reframe event instead?
I talked with @ilmotta about this a while back and it's generally a cleaner patter if we wrap the native-modules in a reframe event as we can avoid the callback hell 😈

Reply:
Agreed! This is a side-effect, so it should go to an effect and be used through an event

Reply:
@J-Son89 I can move it to a reframe event, but I think it would still use a callback (?)

Reply:
Yeah but I think it flattens a bit because we define the callback in events. For this case it won't matter as it's only one callback but if something changes and you need a callback fired in the callback then you will have this issue. 
In any case, I'm pretty sure it should be a reframe event 👌